Server/Server.Core/Server.Core.Static/ServerBuilder.hpp

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <string>
+
+#include "Server.hpp"
+
+namespace Server::Builder
+{
+
+/**
+ * @brief Class builder of Server
+ * @details Class based on builder pattern for creation instances of @link Server @endlink
+ * @todo Think of a more elegant repository initialization interface
+ */
+class ServerBuilder {
+public:
+    ServerBuilder()  = default;
+    ~ServerBuilder() = default;
+
+    inline ServerBuilder& setValue(std::pair<std::string, std::string> keyValue)
+    {
+        _arguments.insert(keyValue);
+        return *this;
+    }
+
+    inline ServerBuilder& setValue(DataAccess::IRepositoryManager* repoManager)
+    {
+        _repository.reset(repoManager);
+        return *this;
+    }
+
+    Server* makeServer() noexcept
+    {
+        auto obj = new Server();
+
+        // form a string with the configuration for connecting to the database
+        std::string options = "";
+        for (const auto& arg : _arguments)
+        {
+            if (arg.first != "--port")
+            {
+                for (size_t i = 0; i < arg.first.length(); ++i)
+                {
+                    if (arg.first[i] != '-')
+                    {
+                        options += arg.first[i];
+                    }
+                }
+                options += "=";
+                options += arg.second;
+                options += " ";
+            }
+        }
+
+        const char* dbOptions = options.c_str();
+        using DataAccess::IAdapter;
+        if (!_repository)
+        {
+            auto up_repo = std::make_unique<DataAccess::PostgreRepositoryManager>(IAdapter::getInstance<DataAccess::PostgreAdapter>(dbOptions));
```

Comment:
Maybe I'm not right, but who do we use postgre instance in server builder? I thought our idead was to be more flexible in the way which repo we wanna use or define what repo we're using when server starts. I can understand it now becayse right now we have only 1 repo ready for use

Reply:
You're right. At the moment, this is some gag so that the server does not break. I commented above the code that we need a more elegant way to make this work.

Reply:
Yeah, I've seen this comment but thought that's not about that point